addons/mrp/models/stock_scrap.py

```diff
@@ -19,15 +19,25 @@ class StockScrap(models.Model):
         domain="[('type', '=', 'phantom'), '|', ('product_id', '=', product_id), '&', ('product_id', '=', False), ('product_tmpl_id', '=', product_template)]",
         check_company=True)
 
-    @api.onchange('workorder_id')
-    def _onchange_workorder_id(self):
-        if self.workorder_id:
-            self.location_id = self.workorder_id.production_id.location_src_id.id
+    @api.depends('workorder_id', 'production_id')
+    def _compute_location_id(self):
+        remaining_scrap = self.browse()
 
-    @api.onchange('production_id')
-    def _onchange_production_id(self):
-        if self.production_id:
-            self.location_id = self.production_id.move_raw_ids.filtered(lambda x: x.state not in ('done', 'cancel')) and self.production_id.location_src_id.id or self.production_id.location_dest_id.id
+        for scrap in self:
+            if scrap.production_id:
+                if scrap.production_id.move_raw_ids.filtered(
```

Comment:
you can check the production_id state instead, the condition will be simpler to understand

Reply:
I've changed it to `if scrap.production_id.state != 'done'`, is this what you meant?